compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -516,6 +559,74 @@ struct QuantizeWeights final : public luci::CircleNodeMutableVisitor<bool>
 
 } // namespace
 
+/** BEFORE
+ *
+ *         [CircleNode]             [CircleNode]
+ *           (qparam1)                (qparam2)
+ *                   \                    /
+ *                    \                  /
+ *                    [CircleConcatenation]
+ *                          (qparam3)
+ *
+ *  AFTER
+ *         [CircleNode]             [CircleNode]
+ *           (qparam3)                (qparam3)
+ *                   \                    /
+ *                    \                  /
+ *                    [CircleConcatenation]
+ *                          (qparam3)
+ */
+void propagate_concat_quantparam(luci::CircleConcatenation *concat)
+{
+  // Check if concat is uint8-quantized
+  if (!is_uint8_quantized(concat))
+    return;
+
+  // Check if concat has no fused activation function
+  if (concat->fusedActivationFunction() != luci::FusedActFunc::NONE)
+    return;
+
+  const auto num_inputs = concat->numValues();
+  for (uint32_t i = 0; i < num_inputs; i++)
+  {
+    auto node = loco::must_cast<luci::CircleNode *>(concat->arg(i));
+
+    // Skip if this input is CONCAT Op
+    if (node->opcode() == luci::CircleOpcode::CONCATENATION)
+      continue;
+
+    // Skip if this input is used by other Ops
+    auto succs = loco::succs(node);
+    if (succs.size() != 1)
+      continue;
+
+    assert(succs.find(concat) != succs.end());
+
+    // Quantize constant values
+    if (node->opcode() == luci::CircleOpcode::CIRCLECONST)
+    {
+      luci::CircleConst *const_node = loco::must_cast<luci::CircleConst *>(node);
+      assert(const_node->dtype() == loco::DataType::FLOAT32);
```

Comment:
Looks redundant in the view of the check on the next line. It could also break a negative test in debug mode (if it relies on throwing an exception).

Reply:
I'll wait for this before merge...

Reply:
> It could also break a negative test in debug mode

@s-barannikov  Agreed. I've removed the check. PTAL